File: 23-3-20/HONG/1/2003.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int main() {

  int N, M;
  cin >> N >> M;

  int arr[10001] = {0};
  cin >> arr[1];
  for (int i = 2; i <= N; i++) {
    cin >> arr[i];
    arr[i] += arr[i - 1];
  }

  int left = 1;
  int right = 1;
  int sum = 0;
  int answer = 0;

  while (left <= N && right <= N) {
    sum = arr[right] - arr[left - 1];
    if (sum == M) {
      answer += 1;
      right += 1;
    } else if (sum < M) {
      right += 1;
    } else if (sum > M) {
      left += 1;
    }
  }

  cout << answer;

  return 0;
}

File: 24-1/HONG/24-1-31/2003.py
N, M = map(int, input().split())
arr = list(map(int, input().split()))
for i in range(1, N):
  arr[i] += arr[i-1]
arr.insert(0,0)

cnt = 0
left, right = 1, 1
while True :

  if right > N or left > N:
    break

  sum = arr[right] - arr[left-1] # [left, right] 사이의 합

  if sum == M:
    cnt += 1
    right += 1
  elif sum < M:
    right += 1
  else:
    left += 1

print(cnt)